src/sentry_core.c

```diff
@@ -288,9 +310,13 @@ sentry_capture_event(sentry_value_t event)
 void
 sentry_handle_exception(sentry_ucontext_t *uctx)
 {
+    const sentry_options_t *opts = sentry_get_options();
+    if (!opts) {
+        return;
+    }
     SENTRY_DEBUG("handling exception");
-    if (g_options->backend && g_options->backend->except_func) {
-        g_options->backend->except_func(g_options->backend, uctx);
+    if (opts->backend && opts->backend->except_func) {
+        opts->backend->except_func(opts->backend, uctx);
```

Comment:
This didn't change, but the signature makes me wonder who takes ownership of `uctx`. Since it's not `const`, it seems like we might need to free in branches where we don't pass it on. Otherwise, can we make it `const` by any chance?

Reply:
nice catch! made it const